sycl/test-e2e/README.md

```diff
@@ -10,19 +11,95 @@
 
 # Overview
 This directory contains SYCL-related tests distributed in subdirectories based
-on testing scope.
+on testing scope. They follow the same style as other LIT tests but have a minor
+but crucial difference in the behavior of certain directives.
+
+First, some background. SYCL end-to-end tests take much longer time to compile
+than they spend on execution on a device. As such, we want to be able to
+structure the test in such a way that we compile it once and then execute
+multiple times on different devices (via `ONEAPI_DEVICE_SELECTOR`) to get
+required test coverage. The issue here is that the standard approach to `RUN`
+directives and substitutions doesn't allow us to expand a line into only
+dynamically known number of commands.
+
+To overcome that, these SYCL end-to-end tests implement a subclass of
+`lit.formats.ShTest` that adjusts default behavior to enable our usage scenario.
+Mainly, we introduce `%{run}` *expansion* that generates multiple commands from
+a single `RUN` directive - one per device in `sycl_devices` parameter (more
+on the parameter below). There is a small number of tests that either need
+multiple devices or target the device selector itself. For such tests we have a
+regular `%{run-unfiltered-devices}` substitution that doesn't set a
+`ONEAPI_DEVICE_SELECTOR` filter nor does it create multiple commands.
+
+This custom LIT test format also overrides the meaning of
```

Comment:
The text above refers to "SYCL end-to-end tests".

> To overcome that, these SYCL end-to-end tests implement a subclass of `lit.formats.ShTest` that adjusts default behavior to enable our usage scenario.

I think there we also should say that we enhanced LIT substitution patterns. "SYCL end-to-end tests" can be misinterpreted.

Reply:
I'm not sure what you're trying to say here. I don't think saying "we enhanced LIT substitution pattern` is accurate because we did no customizations to the generic LIT framework/implementation.

Reply:
Anyway, we should talk about LIT (i.e. **testing framework**) and not the _tests_ itself. I know that LIT allows customizations - it's documented feature, but the way it's phrased in the document says nothing about LIT.

"_SYCL end-to-end tests implement a subclass of `lit.formats.ShTest`_" doesn't sound right to me.